### ISSUE TITLE
lint: add intbool checker

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#18]
Find integer-typed values that only receive 2 values (so, 2 states in total).
Suggest replacing them with `bool`.

This requires being conservative to avoid too much false-positives.

Example:
```go
	x := 0
	if expr() > 10 {
		x = 1
	}
	return x != 0
// consider making `x' a boolean
```
This implies that suggested code could look like:
```go
	x := false
	if expr() > 10 {
		x = true
	}
	return x
```
This whole code can be simplified to `return expr() > 10`, but that's irrelevant.

The checker should also detect struct fields that have integer type, but only used in boolean contexts (`x != 0`, `x == 1`, etc.).

**Comment by Quasilyte** [comment]
https://go-review.googlesource.com/c/go/+/65491

**Comment by Quasilyte** [comment]
Out of scope.
